### PR TITLE
fix django dependency

### LINKED ISSUE #16: Incompatible with Django 3
When i try to install `poetry install` with the **pyproject.toml** above:

```toml
[tool.poetry]
name = "poeta"
version = "0.1.0"
description = ""
authors = ["Urahara"]

[tool.poetry.dependencies]
python = "^3.6"
django = "3.0.8"
django-public-admin = "^0.0.2"

[tool.poetry.dev-dependencies]

[build-system]
requires = ["poetry>=0.12"]
build-backend = "poetry.masonry.api"
```

Result:
```sh
Updating dependencies
Resolving dependencies... (0.4s)

[SolverProblemError]
Because no versions of django-public-admin match >0.0.2,<0.0.3
 and django-public-admin (0.0.2) depends on django (>=2.0.0,<3.0.0), django-public-admin (>=0.0.2,<0.0.3) requires django (>=2.0.0,<3.0.0).
So, because poeta depends on both django (3.0.8) and django-public-admin (^0.0.2), version solving failed.
```

Slightly modifying to use the latest 2.x release of django i able to use without problems:

```toml
[tool.poetry]
name = "poeta"
version = "0.1.0"
description = ""
authors = ["Urahara"]

[tool.poetry.dependencies]
python = "^3.6"
django = "2.2.14"
django-public-admin = "^0.0.2"

[tool.poetry.dev-dependencies]

[build-system]
requires = ["poetry>=0.12"]
build-backend = "poetry.masonry.api"
```

Result:
```sh
Updating dependencies
Resolving dependencies... (2.8s)

Writing lock file


Package operations: 4 installs, 0 updates, 0 removals

  - Installing pytz (2020.1)
  - Installing sqlparse (0.3.1)
  - Installing django (2.2.14)
  - Installing django-public-admin (0.0.2)
```

### PR DESCRIPTION
**Pull request by Urahara**
Close #16 

I run `poetry update` and then `poetry lock`, but i'm not sure if ok to update the `poetry.lock`. Heres the diff: https://gist.github.com/Urahara/1e20b9f00eb740b3faaa75952a7471b1/revisions?diff=split

If that's ok i can commit this.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [tool.poetry]
 name = "django-public-admin"
-version = "0.0.2"
+version = "0.0.3"
```
Comment (cuducos) [inline pyproject.toml]: Can you revert this change and rebase?

I’m really sorry it’s not documented, but I try to keep an (arguably) good practice of only committing version bumps in a specific commit __with__ the proper `CHANGELOG` update.

Usually, I may decide to pack new changes in the new release, and then it makes no sense to bump it here : )

**Review by cuducos — CHANGES_REQUESTED** [review]
Hi, many thanks — it looks like a good catch and fix :tada: 

Before proceeding to merge, I would like to clarify three things:

**1. Version bump**

(as explained in an inline comment)

**2. Update `poetry.lock`**

> I run `poetry update` and then `poetry lock`, but i'm not sure if ok to update the `poetry.lock`.

Yes, you should do that. This is the only way for us to have [deterministic builds](https://en.wikipedia.org/wiki/Reproducible_builds)  : )

**3. How to test it?**

Can you share how you reproduced scenario to test this change really fixes the bug?

File: pyproject.toml (hunk 1 above)
Comment (Urahara) [reply pyproject.toml]: 1 - I'm AFK, so i'm using github web, i will revert in another commit, because i can't rebase right now.
2 -  Done.
3 - After this change i try install both Django 2 and 3 and works!

**Comment by cuducos** [comment]
> 3 - After this change i try install both Django 2 and 3 and works!

I'm sorry if I wasn't clear, my bad.

What I meant wasn't _whether you have tested it or not_ but _how have you tested_. For example: from source, what command, from GitHub or from a “fake” PyPI release, etc.

**Comment by Urahara** [comment]
> > 3 - After this change i try install both Django 2 and 3 and works!
> 
> I'm sorry if I wasn't clear, my bad.
> 
> What I meant wasn't _whether you have tested it or not_ but _how have you tested_. For example: from source, what command, from GitHub or from a “fake” PyPI release, etc.

Hmm...

- I cloned repo
- Initialized a `pipenv` and install `poetry`
- Then i edit the version of django in `toml`
- Run `poetry install`

```sh
(gh) django-public-admin|master ⇒ poetry install
Installing dependencies from lock file


Package operations: 46 installs, 1 update, 0 removals

  - Updating importlib-metadata (1.1.3 -> 1.7.0)
  - Installing markupsafe (1.1.1)
  - Installing pytz (2020.1)
  - Installing alabaster (0.7.12)
  - Installing appdirs (1.4.4)
  - Installing babel (2.8.0)
  - Installing click (7.1.2)
  - Installing distlib (0.3.1)
  - Installing docutils (0.16)
  - Installing filelock (3.0.12)
  - Installing imagesize (1.2.0)
  - Installing importlib-resources (3.0.0)
  - Installing jinja2 (2.11.2)
  - Installing mccabe (0.6.1)
  - Installing more-itertools (8.4.0)
  - Installing packaging (20.4)
  - Installing pathspec (0.8.0)
  - Installing pluggy (0.13.1)
  - Installing py (1.9.0)
  - Installing pycodestyle (2.6.0)
  - Installing pyflakes (2.2.0)
  - Installing pygments (2.6.1)
  - Installing regex (2020.7.14)
  - Installing snowballstemmer (2.0.0)
  - Installing sphinxcontrib-applehelp (1.0.2)
  - Installing sphinxcontrib-devhelp (1.0.2)
  - Installing sphinxcontrib-htmlhelp (1.0.3)
  - Installing sphinxcontrib-jsmath (1.0.1)
  - Installing sphinxcontrib-qthelp (1.0.3)
  - Installing sphinxcontrib-serializinghtml (1.1.4)
  - Installing toml (0.10.1)
  - Installing typed-ast (1.4.1)
  - Installing wcwidth (0.2.5)
  - Installing asgiref (3.2.10)
  - Installing black (19.10b0)
  - Installing flake8 (3.8.3)
  - Installing pytest (5.4.3)
  - Installing sphinx (2.4.4)
  - Installing sqlparse (0.3.1)
  - Installing virtualenv (20.0.27)
  - Installing django (3.0.8)
  - Installing pytest-black (0.3.10)
  - Installing pytest-flake8 (1.0.6)
  - Installing pytest-mock (2.0.0)
  - Installing readthedocs-sphinx-ext (1.0.4)
  - Installing sphinx-rtd-theme (0.4.3)
  - Installing tox (3.18.0)
  - Installing django-public-admin (0.0.2)
```

Then tried to install version 2.x:

```sh
(gh) django-public-admin|master ⇒ poetry add django==2.2.14

Updating dependencies
Resolving dependencies... (0.3s)

Writing lock file


Package operations: 0 installs, 1 update, 1 removal

  - Updating django (3.0.8 -> 2.2.14)
  - Removing asgiref (3.2.10)
```

**Comment by Urahara** [comment]
Sorry if i'm testing wrong, i never used poetry only pipenv.

**Comment by cuducos** [comment]
> Sorry if i'm testing wrong, i never used poetry

Worry not — I am also new to Poetry. BTW that is why I asked how you tested.

I think it makes sense what you did TBH. Yet, I would like to corroborate that with another testing strategy, to make sure it _also_ works with an external dependency. Having the edited version of _Django Public Admin_ at, let’s say, `/home/cuducos/django-public-admin` I would

1. start two new projects with Poetry, one depending on Django 2 and the other, on Django 3
2. add to both the _local_ version of _Django Public Admin_ with something like `poetry add /home/cuducos/django-public-admin` (not sure if this works, but [there’s a discussion about that for us to find some references](https://github.com/python-poetry/poetry/issues/1135))

As soon as I can test this I can squash and merge your commits to the main branch  : )

Once more, many many thanks, buddy!

**Comment by Urahara** [comment]
> > Sorry if i'm testing wrong, i never used poetry
> 
> Worry not — I am also new to Poetry. BTW that is why I asked how you tested.
> 
> I think it makes sense what you did TBH. Yet, I would like to corroborate that with another testing strategy, to make sure it _also_ works with an external dependency. Having the edited version of _Django Public Admin_ at, let’s say, `/home/cuducos/django-public-admin` I would
> 
> 1. start two new projects with Poetry, one depending on Django 2 and the other, on Django 3
> 2. add to both the _local_ version of _Django Public Admin_ with something like `poetry add /home/cuducos/django-public-admin` (not sure if this works, but [there’s a discussion about that for us to find some references](https://github.com/python-poetry/poetry/issues/1135))
> 
> As soon as I can test this I can squash and merge your commits to the main branch : )
> 
> Once more, many many thanks, buddy!

I tried your approach and didn't work, only way i can think of is creating a package in pypi.

**Comment by cuducos** [comment]
> I tried your approach and didn't work, only way

There are different approaches in the issue I linked. None of them actually works? 

There should be a way to test it without PyPI, I'll explore this issue over the weekend, ok?

**Comment by Urahara** [comment]
I tested with **wheel**, I created those folders:

- django-public-admin (mine branch)
- django2
- django3

Run `poetry build` on **django-public-admin** copy whl file to django(2|3) folders, then installed:

```sh
# pipenv run pip install django_public_admin-0.0.2-py3-none-any.whl
Processing ./django_public_admin-0.0.2-py3-none-any.whl
Requirement already satisfied: django>=2 in /root/.local/share/virtualenvs/django2--pPR18ve/lib/python3.6/site-packages (from django-public-admin==0.0.2) (2.2.14)
Requirement already satisfied: sqlparse>=0.2.2 in /root/.local/share/virtualenvs/django2--pPR18ve/lib/python3.6/site-packages (from django>=2->django-public-admin==0.0.2) (0.3.1)
Requirement already satisfied: pytz in /root/.local/share/virtualenvs/django2--pPR18ve/lib/python3.6/site-packages (from django>=2->django-public-admin==0.0.2) (2020.1)
Installing collected packages: django-public-admin
Successfully installed django-public-admin-0.0.2
# cd ..
# ls
django-public-admin  django2  django3
# cd django3
# pipenv run pip install django_public_admin-0.0.2-py3-none-any.whl
Processing ./django_public_admin-0.0.2-py3-none-any.whl
Requirement already satisfied: django>=2 in /root/.local/share/virtualenvs/django3-bLZdFYze/lib/python3.6/site-packages (from django-public-admin==0.0.2) (3.0.8)
Requirement already satisfied: sqlparse>=0.2.2 in /root/.local/share/virtualenvs/django3-bLZdFYze/lib/python3.6/site-packages (from django>=2->django-public-admin==0.0.2) (0.3.1)
Requirement already satisfied: asgiref~=3.2 in /root/.local/share/virtualenvs/django3-bLZdFYze/lib/python3.6/site-packages (from django>=2->django-public-admin==0.0.2) (3.2.10)
Requirement already satisfied: pytz in /root/.local/share/virtualenvs/django3-bLZdFYze/lib/python3.6/site-packages (from django>=2->django-public-admin==0.0.2) (2020.1)
Installing collected packages: django-public-admin
Successfully installed django-public-admin-0.0.2
```

I did the same using the master branch from our repo:

```sh
root@108a19493226:/tests/teste/django3# pipenv run pip install django_public_admin-0.0.2-py3-none-any.whl 
Processing ./django_public_admin-0.0.2-py3-none-any.whl
Collecting django<3.0.0,>=2.0.0
  Using cached Django-2.2.14-py3-none-any.whl (7.5 MB)
Requirement already satisfied: pytz in /root/.local/share/virtualenvs/django3-aAlVc99F/lib/python3.6/site-packages (from django<3.0.0,>=2.0.0->django-public-admin==0.0.2) (2020.1)
Requirement already satisfied: sqlparse>=0.2.2 in /root/.local/share/virtualenvs/django3-aAlVc99F/lib/python3.6/site-packages (from django<3.0.0,>=2.0.0->django-public-admin==0.0.2) (0.3.1)
Installing collected packages: django, django-public-admin
  Attempting uninstall: django
    Found existing installation: Django 3.0.8
    Uninstalling Django-3.0.8:
      Successfully uninstalled Django-3.0.8
Successfully installed django-2.2.14 django-public-admin-0.0.2
```

**Comment by cuducos** [comment]
I tested not with the wheel, but with the directory — [as described in my previous comment](https://github.com/cuducos/django-public-admin/pull/17#issuecomment-663247964). It works : )

**Comment by Urahara** [comment]
Strange... but thanks!!
